sonar-db/src/main/java/org/sonar/db/charset/MssqlCharsetHandler.java

```diff
@@ -46,97 +42,81 @@
   private static final String BIN = "BIN";
   private static final String BIN2 = "BIN2";
 
-  protected MssqlCharsetHandler(SqlExecutor selectExecutor) {
+  private final Metadata metadata;
+
+  MssqlCharsetHandler(SqlExecutor selectExecutor) {
     super(selectExecutor);
+    this.metadata = new Metadata();
```

Comment:
hum... this is bordeline
as it is right now, it seems the code has only been altered solely for testing

I would favor code where `Metadata` is never instanciated in `MssqlCharsetHandler` but rather injected all the time. As so, `Metadata` would cleanly be a dependency of `MssqlCharsetHandler`.